java/algorithm/src/test/java/org/apache/arrow/algorithm/search/TestParallelSearcher.java

```diff
@@ -63,16 +95,19 @@ public void testParallelIntSearch() throws ExecutionException, InterruptedExcept
       targetVector.allocateNew(VECTOR_LENGTH);
       keyVector.allocateNew(VECTOR_LENGTH);
 
+      VectorValueComparator<IntVector> comparator = comparatorType == ComparatorType.EqualityComparator ? null
+          : DefaultVectorComparators.createDefaultComparator(targetVector);
```

Comment:
Could you add a comment here, I forget the different between order comparator and default comparator?  I'm not really clear what this test is demonstrating?

Reply:
@emkornfield Comments added. Thanks for your feedback. More comments are added to the API Java doc.

Please let me give some background here. We have two ways to compare elements in vectors:
1. Equality semantics: based on `RangeEqualsVisitor`. For an element-wise comparison, we have two possible results: equal/non-equal.
2. Ordering semantics: based on `ValueVectorComparator`. For an element-wise comparison, we have three possible results: less than, equal to, and greater than.

Previously, we supported the equal semantics in this class, now we add support for the ordering semantics.

We basically reuse the original test cases, but transform it to a parameterized test class, so the new algorithm can be tested.